libflux/src/semantic/types.rs

```diff
@@ -296,8 +294,8 @@ impl Substitutable for MonoType {
             | MonoType::String
             | MonoType::Duration
             | MonoType::Time
-            | MonoType::Regexp => self,
-            MonoType::Var(tvr) => sub.apply(tvr),
+            | MonoType::Regexp => self.clone(),
```

Comment:
Yeah, but this clone seems like trivial because  the enum has no parameter. I was worried of cloning complex monotypes as  `Function` that can recurse.